receiver/iisreceiver/scraper.go

```diff
@@ -0,0 +1,91 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+// +build windows
+
+package iisreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/iisreceiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/winperfcounters"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/iisreceiver/internal/metadata"
+)
+
+type iisReceiver struct {
+	params   component.ReceiverCreateSettings
+	config   *Config
+	consumer consumer.Metrics
+	watchers []winperfcounters.PerfCounterWatcher
+}
+
+// new returns a iisReceiver
+func newIisReceiver(params component.ReceiverCreateSettings, cfg *Config, consumer consumer.Metrics) *iisReceiver {
+	return &iisReceiver{params: params, config: cfg, consumer: consumer}
+}
+
+// Start creates and starts the prometheus receiver.
+func (rcvr *iisReceiver) start(ctx context.Context, host component.Host) error {
+	rcvr.watchers = []winperfcounters.PerfCounterWatcher{}
+	for _, objCfg := range getScraperCfgs() {
+		objWatchers, err := objCfg.BuildPaths()
+		if err != nil {
+			rcvr.params.Logger.Warn("some performance counters could not be initialized", zap.Error(err))
```

Comment:
Use the `scrapererror` pattern instead of logging. [Example](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f430ca84115a06af08be846ecfd3a234dadc5944/receiver/couchdbreceiver/metrics.go#L30)

Reply:
Updated